docs/library/drivers/common/watchdog/index.rst

```diff
@@ -3,3 +3,107 @@
+design under test to be able to configure the watchdog module properly. The
+watchdog object can be used on the entire test stimulus or on smaller test
+cases, where there is a moderate chance that the system may hang, and to
+prevent such instances to happen.
+
+Variables
+-------------------------------------------------------------------------------
+
+None are available for direct external access.
+
+Functions
+-------------------------------------------------------------------------------
+
+function new(bit [31:0] timer, string message);
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Creates the watchdog object. The timer variable is used to set the default
+timer value after which the watchdog is triggered. This value must be set in
+nanoseconds. The message value sets the default output message when the
+watchdog is triggered.
+
+function void update_message(string message);
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Used to update the timer value. The timer variable is used to set the new timer
+value after which the watchdog is triggered.
+
+function void update_timer(bit [31:0] timer);
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Used to update the watchdog message. The message value sets the watchdog output
+message when this is triggered. This value must be set in nanoseconds.
+
+task start();
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Starts the watchdog timer in a separate thread. This allows other code to be
+run in parallel with it. When the watchdog timer reaches the wait time that was
+set prior to starting of the watchdog, either with a default value or an
+updated value, it is triggered, prints out the watchdog message and then the
+simulation is halted. If the watchdog receives a stop event, the thread is
+killed and the counting stops. This means that the watchdog will not trigger in
+this instance and the simulation will not be halted.
+
+task stop();
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Stops the watchdog timer when called. Used when the designer wants to stop the
+watchdog timer.
+
+task reset();
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Stops and then starts again the watchdog timer using the start and stop
+functions.
+
+Usage and recommendations
+-------------------------------------------------------------------------------
+
+Basic usage of the watchdog timer:
+
+* Declare the watchdog timer, give an initial value for the timer and the
+  message. These can be changed later
+* Update the watchdog timer value and message if needed
+* Start the watchdog timer
+* Add test stimulus that needs to be timed by the watchdog timer
+* Stop the watchdog timer if the stimulus finished the process
+
+Important notes for the watchdog timer:
```

Comment:
If you want to emphasize these notes, you can use .. important::

Example of usage:
https://github.com/analogdevicesinc/hdl/blob/dddb39b21f6edd57828d83dbe5ddc454330f9317/docs/library/axi_adc_decimate/index.rst?plain=1#L14C1-L14C15